### PR TITLE
Offline history loading

### PR DESCRIPTION
**Pull request by goodboy**
Adds `ib` offline (market is currently closed - rt quotes aren't happening) history loading.
This brings in a bunch of commits factored out of #289 and some manual work for windows.

Please feel free to test for those users that are on this backend.

---
Oh, right because we depended on it, the factored commits also included display loop re-works from #289 which allows manually triggering and update cycle with the original motivation being for historical prepends when the market is not open (so updates from rt quotes aren't triggering updates).

> factor the "graphics draw cycle" steps into a sync function that can be triggered (remotely) to update on certain types of events for example when an fsp running in a subactor makes an update but the market is not open - it needs to tell the chart app to update graphics

### CODE REVIEW

File: piker/ui/_display.py [hunk 1]
```diff
@@ -109,6 +111,33 @@ def chart_maxmin(
     return last_bars_range, mx, max(mn, 0), mx_vlm_in_view
 
 
+@dataclass
+class DisplayState:
```
Comment (guilledk) [inline piker/ui/_display.py:115]: Love this new `dataclass`

**Review by guilledk — APPROVED** [review]

File: piker/ui/_display.py (hunk 1 above)
Comment (goodboy) [reply piker/ui/_display.py:115]: Yeah makes it super nice to manual updates - probably will work for remote control decent down the road as well.

File: piker/ui/_display.py [hunk 2]
```diff
@@ -231,222 +281,263 @@ async def graphics_update_loop(
...
+
+def graphics_update_cycle(
+    ds: DisplayState,
+    wap_in_history: bool = False,
+    trigger_all: bool = False,  # flag used by prepend history updates
+
+) -> None:
 
+    # TODO: eventually optimize this whole graphics stack with ``numba``
+    # hopefully XD
```
Comment (iamzoltan) [inline piker/ui/_display.py:296]: heh, I hope so

**Review by iamzoltan — APPROVED** [review]
looks great, can't wait to get something like this for good ol' phil

**Pull request merged**